src/include/sof/audio/component.h

```diff
@@ -602,6 +602,11 @@ struct comp_dev {
 	/* private data - core does not touch this */
 	void *priv_data;	/**< private data */
 
+	/* store frame_fmt to avoid repeated calculation */
+	enum sof_ipc_frame in_frame_fmt;
+	enum sof_ipc_frame in_valid_fmt;
+	enum sof_ipc_frame out_frame_fmt;
+	enum sof_ipc_frame out_valid_fmt;
```

Comment:
I don't see the calculations done on actual hot paths (in the processing calls). Does it makes sense to change SRAM to saved MCPS when the impact is only on the prepare/init paths?

Reply:
this patch purpose is not for save mcps, it target for avoid repeated calculation in copier and mixin mixout for format, only these two modules, other modules does not have format conversion, so I did not change other modules.

Reply:
@btian1  Not following. By adding new fields to component.h, we increase the SRAM use for all components in the system, so if this does not save MCPS, why get rid of the calculations?

Saving in code size can help reduce SRAM consumption as well, but when augmenting widely used structs (like in component.h), we need to ensure we have a good compromise.

Reply:
yes, performance/memory/code size always like a triangle, this adding is reasonable and acceptable(one int), since it is not changing at runtime.

Reply:
@btian1 Ack but I'm arguing that this tradeoff is not good if the code saving is not on the hot path (in a processing function) but the memory cost affects everything.